server/models/conversation.js

```diff
@@ -0,0 +1,27 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const ConversationSchema = new Schema({
+  timeStarted: {
+    type: Date,
+    default: Date.now,
+    required: true
+  },
+  authorOne: {
+    type: Schema.Types.ObjectID,
+    ref: "User",
+    required: true,
+    index: true
+  },
+  authorTwo: {
+    type: Schema.Types.ObjectID,
+    ref: "User",
+    required: true,
+    index: true
+  },
+  messages: [{ type: Schema.Types.ObjectId, ref: "Message" }]
```

Comment:
I'm fine with messages being embedded, but keep in mind that in production this would be bad, since two users can potentially exchange millions of messages, all of which would be returned at once. There's also that fact that a mongo document has a hard size limit of 16mb

Reply:
As this is a discreet Conversation between two people I, perhaps not being overly verbose, never considered that this could get in the millions of messages.  It seems relatively trivial to put a cap on the number of messages stored.  If you think a cap is more difficult than my guess or that there's a better solution I'm ready to research it.

Reply:
Oh, I don't mean a cap. The actual hard limit assuming 150 characters per message and some metadata would be around 80k messages, so nothing you'd have to worry about given the scope of this app lol. Just pointing out that in production you'd want to use something like the hybrid approach here: https://docs.mongodb.com/ecosystem/use-cases/storing-comments/ 

On the routes though, I'd like to see some pagination to keep user experience smooth on the frontend and let us implement infinite scrolling, so I wanted you to look into the _$slice_ operator

Reply:
Fair enough.  Thanks again for pointing me in the right direction.